### ISSUE TITLE
Testing

### ISSUE THREAD

**Issue by djcroissant** [djcroissant/getyrbeta#6]
Want to write tests?? First of all, you're amazing! Life will reward you at every turn. Your glory will be remembered forever. 

Lots of testing to be added. To see where, run coverage to generate an HTML coverage report:

$ coverage run manage.py test
$ coverage html
$ open htmlcov/index.html

**Comment by mraza007** [comment]
Hey Can I work on this issue

**Comment by djcroissant** [comment]
Absolutely! Sorry about the delay. I was out of town over the weekend. What is your familiarity with Django and open source contributions? I am somewhat new, but this has been a really fun project to work on!

**Comment by mraza007** [comment]
Its totally fine I have worked with Django before and I am currently working with Django right now.
I am somewhat new to writing tests maybe I can try and I have done some open source contributions in past so I am familiar with git

**Comment by djcroissant** [comment]
The two areas that are top priority for testing are:
1. Integration tests. There is currently only unit testing. 
2. Unit testing for tripplan/trips/views.py

Or anything else that you are interested. More tests are always welcome.

**Comment by mraza007** [comment]
I can definitely work with you on the web app 
It looks interesting and in the longer run it will help me brush up my Django skills

**Comment by mraza007** [comment]
Can I update the installation docs for linux too

**Comment by mraza007** [comment]
Hey  I will start on this repo on the weekend 
Btw whats the sole purpose of this web app and what makes it cool

**Comment by djcroissant** [comment]
Hi,

Updates to the docs are always welcome! I have been working with React in
JavaScript for the last couple days. Just got my first couple apps built
out. Do you have any experience combining a JavaScript/React front end with
a Django backend? Just curious.

The purpose of this app is to facilitate trip planning for outdoor
adventures. It has also served as an excellent platform for learning
Python/Django. What makes it cool? It takes a process that people (going on
hikes) have to complete frequently and makes it both easier and more
effective. Most people use a spreadsheet, which has it's own collaboration
issues. The app makes it super easy to join a trip and collaborate on trip
planning.

There is a lot of space to leave your impact. Testing is a great place to
start. As you get familiar, feel free to suggest other areas you'd like to
change as well.

**Comment by mraza007** [comment]
I mean I have experience working with vanilla javascript but I haven't touch react since I am currently working in the IOT world. And Plus I believe working this project will help me brush up my skills.
 btw I looked at the front end its really simple plain html/css and javascript maybe we can make the UI better.
These are my thoughts on the project

**Comment by mraza007** [comment]
Just a quick question what are you using for the deployment and I was wondering if we can have a better communication medium or you would just prefer github

**Comment by djcroissant** [comment]
I
​ created a Slack channel so we can chat more easily. You can join here:

https://join.slack.com/t/getyrbeta/shared_invite/enQtMzkwMDA2Njg2MDMyLWFjMjFiN2Q2MDZlMGVhNGI4MWI4NWQ0YWFhNThjNjk1MjEwNmExOWUwNzhiNGQ1ZmMzMDczMjkyYzkxZGIzNzA

For dependencies: I have used a virtualenv. But I agree, something like
pipenv would be an improvement. Especially combined with Docker. That said,
it hasn't bubbled to the top of the priority list since the virtualenv
combined with the requirements files has worked well up to this point.


For the front end: I see the UX design as the final hurtle for getting the
app ready for users. It is something I am actively learning. It is no small
undertaking though. My current plan has been to improve my javascript
through other projects, then come back to GetYrBeta and update the front
end.

Objective: My two main objectives with Get Yr Beta have been 1. To improve
my coding skills and build something I can show in my portfolio; 2. To
create an App that is appealing enough to attract a large user base.
 My prioritization has reflected these objectives. So if there is something
you are interested in learning/improving and you would like to use Get Yr
Beta as a platform, then that can be top priority. It doesn't necessarily
have to be the thing that is top priority for users. However, I believe
that to accomplish the 2nd objective of building a user base, the UX is
currently top priority.


I hope this helps give you an idea of where I'm at. I'm excited that you're
interested in the project!

Derek​

**Comment by mraza007** [comment]
Sure I get your what you are saying
Yeah we definitely tweak UI later 
I have joined slack
